src/com/sap/piper/Utils.groovy

```diff
@@ -72,67 +75,28 @@ def unstashAll(stashContent) {
-            return
-        }
+        parameters.actionName = parameters.get('actionName') ?: 'Piper Library OS'
+        parameters.eventType = parameters.get('eventType') ?: 'library-os'
+        parameters.jobUrlSha1 =  generateSha1(env.JOB_URL)
+        parameters.buildUrlSha1 = generateSha1(env.BUILD_URL)
 
-        def swaCustom = [:]
-
-        /* SWA custom parameters:
-            custom3 = step name (passed as parameter step)
-            custom4 = job url hashed (calculated)
-            custom5 = build url hashed (calculated)
-            custom10 = stage name
-            custom11 = step related parameter 1 (passed as parameter stepParam1)
-            custom12 = step related parameter 2 (passed as parameter stepParam2)
-            custom13 = step related parameter 3 (passed as parameter stepParam3)
-            custom14 = step related parameter 4 (passed as parameter stepParam4)
-            custom15 = step related parameter 5 (passed as parameter stepParam5)
-        */
-
-        def swaUrl = 'https://webanalytics.cfapps.eu10.hana.ondemand.com/tracker/log'
-        def action_name = 'Piper Library OS'
-        def idsite = '827e8025-1e21-ae84-c3a3-3f62b70b0130'
-        def url = 'https://github.com/SAP/jenkins-library'
-        def event_type = parameters.get('eventType') ?: 'library-os'
-
-        swaCustom.custom3 = parameters.get('step')
-        swaCustom.custom4 = generateSha1Inline(env.JOB_URL)
-        swaCustom.custom5 = generateSha1Inline(env.BUILD_URL)
-        swaCustom.custom10 = parameters.get('stageName')
-        swaCustom.custom11 = parameters.get('stepParam1')
-        swaCustom.custom12 = parameters.get('stepParam2')
-        swaCustom.custom13 = parameters.get('stepParam3')
-        swaCustom.custom14 = parameters.get('stepParam4')
-        swaCustom.custom15 = parameters.get('stepParam5')
-
-        def options = []
-        options.push("-G")
-        options.push("-v \"${swaUrl}\"")
-        options.push("--data-urlencode \"action_name=${action_name}\"")
-        options.push("--data-urlencode \"idsite=${idsite}\"")
-        options.push("--data-urlencode \"url=${url}\"")
-        options.push("--data-urlencode \"event_type=${event_type}\"")
-        for(def key : ['custom3', 'custom4', 'custom5', 'custom10', 'custom11', 'custom12', 'custom13', 'custom14', 'custom15']){
-            if (swaCustom[key] != null) options.push("--data-urlencode \"${key}=${swaCustom[key]}\"")
-        }
-        options.push("--connect-timeout 5")
-        options.push("--max-time 20")
-
-        sh(returnStatus: true, script: "#!/bin/sh +x\ncurl ${options.join(' ')} > /dev/null 2>&1 || echo '[${parameters.get('step')}] Telemetry Report to SWA failed!'")
-
-    } catch (MissingContextVariableException noNode) {
-        echo "[${parameters.get('step')}] Telemetry Report to SWA skipped, no node available!"
+        Telemetry.notify(this, config, parameters)
```

Comment:
For us it would be good that parameters is a map with named keys. So currently there is no simple way to extract the e.g. the stageDuration. 
Would like to do parameters.stageDuration and currently the value is stored in stepParamKey3, but that might change?

Reply:
Would this be sufficient?

https://github.com/SAP/jenkins-library/pull/442/files#diff-107e29747c644fec9fcc5b5dac73bff6R113

The map has still "stepParamX" but also matching "stepParamKeyX" values.

We need to have the mapping somewhere, currently it's in the steps itself. We could now discuss if this is of some more SWA specific kind and should be part of the specific listener.

Reply:
We just added the parameters from `stageWrapper` additionally as named parameter.